### PR TITLE
Do not call exit() on MSVC Port when calling vPortEndScheduler

### PR DESCRIPTION
**Pull request by Ju1He1**
Description
-----------
See description in issue
https://github.com/FreeRTOS/FreeRTOS-Kernel/issues/623

Test Steps
-----------
Call vPortEndScheduler() instead of just ending the scheduler the problem is completely stopped

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aggarg** [comment]
Would you please tell us your use case and the problem that you are trying to solve. Currently you are adding a check `pdTRUE == xPortRunning` and `xPortRunning` is nowhere set to `false` which essentially make this loop an infinite loop.

**Comment by Ju1He1** [comment]
hey @aggarg 
thx fo your feedback. This was a copy & paste error. I'm working with an older FreeRTOS version locally. 
xPortRunning  is not set to false when calling vPortEndScheduler

In our use case we have some modules which depend on FreeRTOS components. All our modules (FreeRTOS and non FreeRTOS Modules) are tested using GTest


When testing modules which depend on FreeRTOS we are running GTest from a FreeRTOS task (i.e. the Freertos scheduler has been started). Obviously Gtest Tests can fail, so we have to return a result.
In the posix port this works well, however in the Windows Port vPortEndScheduler() always calls exit(0) which kinda breaks the purpose of unit tests ;).
This Merge Request trys to make the MSVC pot behave like the Posix Port. It does not make sense to me, why the Posix Port behaves differently than the MSVC port. ;)


our unit test mains look like this.
```

int main(int argc, char* argv[])
{
::testing::InitGoogleTest(&argc, argv);


  configASSERT(pdTRUE == xTaskCreate(
    RunGTestInTask,       /* Function that implements the task. */
    "GTestTask",          /* Text name for the task. */
    200,      /* Stack size in words, not bytes. */
    (void*)1,    /* Parameter passed into the task. */
    5,/* Priority at which the task is created. */
    &mg_gtestTask));      /* Used to pass out the created task's handle. */


  /* Start the tasks and timer running. */
  vTaskStartScheduler();
  /*
  in the windows simulation we will never get here
  in the Linux simulation we will get here after vTaskEndScheduler() was called
  */
  exit(mg_iGTestResult);
}


static void RunGTestInTask(void* pvParameters)
{
  mg_iGTestResult = RUN_ALL_TESTS();
  sendShutdownRequestToAllTasks();
  waitForAllTasksToFinnish();

  vTaskEndScheduler();


#ifdef _MSC_VER
  exit(mg_iGTestResult);
#endif // _MSC_VER
}
```

**Comment by RichardBarry** [comment]
Looks like this is in hand - so just a reminder to check this works with GCC too.

File: portable/MSVC-MingW/port.c [hunk 1]
```diff
@@ -160,7 +160,7 @@ TIMECAPS xTimeCaps;
     /* Just to prevent compiler warnings. */
     ( void ) lpParameter;
 
-    for( ;; )
+    while(pdTRUE == xPortRunning)
```
Comment (aggarg) [inline portable/MSVC-MingW/port.c:163]:
```suggestion
    while( xPortRunning == pdTRUE )
```

File: portable/MSVC-MingW/port.c [hunk 2]
```diff
@@ -176,33 +176,32 @@ TIMECAPS xTimeCaps;
         {
             Sleep( portTICK_PERIOD_MS );
         }
+        if(pdTRUE == xPortRunning)
```
Comment (aggarg) [inline portable/MSVC-MingW/port.c:179]:
```suggestion

        if( xPortRunning == pdTRUE )
```

**Comment by aggarg** [comment]
Thank you for the detailed explanation. Just left some cosmetic suggestions to keep the style consistent.

**Comment by Ju1He1** [comment]
> Thank you for the detailed explanation. Just left some cosmetic suggestions to keep the style consistent.

Thx I have added your suggestions to my Request. :)

**Review by aggarg — APPROVED** [review]

**Review by kar-rahul-aws — APPROVED** [review]

**Pull request merged**